### ISSUE TITLE
User interface for error reporting

### ISSUE THREAD

**Issue by acdha** [LibraryOfCongress/bagger-js#44]
We currently have some error paths which are either the console log or alert dialogs. We should consolidate those into a better user experience — perhaps something like an info list which is collapsed by default or some per-file badging on the bag contents when applicable.

**Comment by johnscancella** [comment]
We should also add errors when required fields are blank. I didn't realize that the `bucket` field wasn't filled in at first and hitting the `Test configuration` button did nothing.

**Comment by acdha** [comment]
Good point — I was tabling that for upgrading to Bootstrap 4 since that uses the standard HTML5 `:valid` and `:invalid` CSS selectors but it's really easy to add now.

**Comment by acdha** [comment]
See https://github.com/LibraryOfCongress/bagger-js/commit/72885bde3f20e95c43c560d97cddc8e6610e5771

**Comment by johnscancella** [comment]
looks good to me. The only nitpick I have is it is immediately red, but not a showstopper by any means.

**Comment by acdha** [comment]
Yeah, we could catch that but there's a counterargument that it makes it more obvious that these are required values. I also don't know how many people would be entering these values directly versus having most of them prefilled from the URL hash/querystring.

**Comment by acdha** [comment]
With the current state of #47 I think the core file operations are reasonably well covered: errors are indicated and upload errors can be retried. 

I'm not sure there's a sensible way to retry a hash error — maybe that's something like dropping a ton of files into the window and deleting one of them before it can be hashed? The cleanest way to handle that might be by adding some UI to delete an entry if you change your mind about uploading something but don't want to discard your other files, metadata entry, etc. since you could always delete and re-add the file.

**Comment by acdha** [comment]
The last commit on #47 yesterday (https://github.com/LibraryOfCongress/bagger-js/pull/47/commits/751cf4c98f4d1ecc7b6823150861de0e1243bf87) added basic removal support including attempting to delete the file from S3.
